helpers/test/test_system_interaction.py

```
@@ -69,6 +70,35 @@ def test7(self) -> None:
         act = re.sub(r"rc='(\d+)'", "rc=''", act)
         self.check_string(act)
 
+    def test8(self) -> None:
+        """
+        Test .system() on a failing cmd with --tee switch enabled and no other parameters.
+        """        
+        with self.assertRaises(AssertionError) as cm:
+            hsystem.system("ls this_should_fail", tee=True)    
+        
+    
+    def test9(self) -> None:
+        """
+        Test .system() on a failing cmd with --tee switch enabled and specified log output file.
+        """  
+        log_dir = self.get_scratch_space()
+        log_file_path = os.path.join(log_dir, "tee_log")   
+        #Creates AssertRaisesContext error (seems unexpected)
+        with self.assertRaises(RuntimeError) as cm:
+            hsystem.system("ls this_should_fail", tee=True, output_file=log_file_path)
+        # Check that log file contains error message.
+        log_content = hio.from_file(log_file_path)
+        self.check_string(log_content)
+
+    def test10(self) -> None:
+        """
+        Test .system() on a failing cmd with --tee switch and allowedErrors enabled.
+        """
+        with self.assertRaises(TypeError) as cm:
+            hsystem.system("ls this_should_fail", tee=True, allowedErrors=True)
```

Well, this one should pass actaully
but see `system()` func interface - it does not have `allowedErrors` param
see  the func interface and https://github.com/sorrentum/sorrentum/issues/355#issuecomment-1616780206

Also do 
```
rc = hsystem.system(...
```
and then assert the value of `rc`

---

In the `test9` unit test errors are not permitted and therefore the command should raise an error, correct? If so I am confused as to why I am trying to get the returned value `rc` from the failed `hsystem.system()` method because this method should not return anything due to the exception raised within it. I only ask because I want to make sure I am understanding the scenario correctly.

---

yep, you're right, in this case `rc` should not be set, since cmd fails
so no reason to do `rc = ` here
it is needed in the next test though